libcudacxx/include/cuda/std/__type_traits/is_callable.h

```diff
@@ -35,6 +36,21 @@ struct __is_callable : _IsValidExpansion<__call_result_t, _Func, _Args...>
 template <class _Func, class... _Args>
 inline constexpr bool __is_callable_v = _IsValidExpansion<__call_result_t, _Func, _Args...>::value;
 
+namespace detail
+{
+template <class _Func, class... _Args>
+using __if_nothrow_callable_t _CCCL_NODEBUG_ALIAS =
+  _CUDA_VSTD::enable_if_t<noexcept(_CUDA_VSTD::declval<_Func>()(_CUDA_VSTD::declval<_Args>()...))>;
+} // namespace detail
+
+template <class _Func, class... _Args>
+struct __is_nothrow_callable : _IsValidExpansion<detail::__if_nothrow_callable_t, _Func, _Args...>
+{};
```

Comment:
Question: Is this ever used now that we require C++17?

Reply:
i don't understand. i'm adding it because i need it. why does c++17 matter?